### PR TITLE
B 20010 int

### PR DESCRIPTION
**Pull request by r-mettler**
## Summary

This backlog adds a new Headquarters Office user role. Further downstream tasks will add privileges meant for the HQ role which will limit pages to read only.

Is there anything you would like reviewers to give additional scrutiny?

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login to the admin application
2. Create a new Office user
3. Assign the Headquarters role and TOO role to the user and save
image (2)

4. Go to the Office app and sign in as the newly created user
5. Click the change user role and verify you can login with the HQ role
image (3)
NOTE: At this point you should be able to do what a TOO can do but this story was to just allow a new login role

6. On the Office App login screen click the Request Account button in the top right corner
7. Fill out the form
8. Verify there is a checkbox for Headquarters role and check it
9. Click Submit and make sure it submits the request
10. Login to the Admin App under the Requested Office Users section find your requested account and approve it
11. Go to the Office App and verify you can login to the app with your new requested user

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

**Comment by ajlusk** [comment]
Either way I create an HQ User something seems to be missing.

When I create an HQ user as the testing instructions suggest, via the admin app, I am unable to log in as the new user. I get a Forbidden error. The error message says the account is not active, however `users.active` is `true`. The user is missing a value in the `users.okta_id` column. When I added a made up ID there the user works as described. This appears to happen for all users created in the admin app so could be an issue on my end. 

When I create and HQ user from `http://officelocal:3000/devlocal-auth/login`, the user is only assigned just the TOO role (verified in the DB) rather than the HQ role or both. This method of user creation is not listed in the testing instructions but seemed relevant to test.

**Comment by r-mettler** [comment]
> Either way I create an HQ User something seems to be missing.
> 
> When I create an HQ user as the testing instructions suggest, via the admin app, I am unable to log in as the new user. I get a Forbidden error. The error message says the account is not active, however `users.active` is `true`. The user is missing a value in the `users.okta_id` column. When I manually added a uuid there the user worked as expected. This appears to happen for all users created in the admin app so could be an issue on my end.
> 
> When I create and HQ user from `http://officelocal:3000/devlocal-auth/login`, the user is only assigned just the TOO role (verified in the DB) rather than the HQ role or both. This method of user creation is not listed in the testing instructions but seemed relevant to test.

Looking into it, I had to make some changes for a test to pass, so its possible it broke something else when the test pass. 😞

**Comment by r-mettler** [comment]
> Either way I create an HQ User something seems to be missing.
> 
> When I create an HQ user as the testing instructions suggest, via the admin app, I am unable to log in as the new user. I get a Forbidden error. The error message says the account is not active, however `users.active` is `true`. The user is missing a value in the `users.okta_id` column. When I manually added a uuid there the user worked as expected. This appears to happen for all users created in the admin app so could be an issue on my end.
> 
> When I create and HQ user from `http://officelocal:3000/devlocal-auth/login`, the user is only assigned just the TOO role (verified in the DB) rather than the HQ role or both. This method of user creation is not listed in the testing instructions but seemed relevant to test.

Its a known issue that the users are missing okta_id when created in the admin app. I resolved the Forbidden error though so you should login.

File: migrations/app/migrations_manifest.txt [hunk 1]
```diff
@@ -924,9 +924,11 @@
 20240411201158_add_application_parameter_and_validation_code_table.up.sql
 20240412201837_edit_gun_safe_entitlement_not_null.up.sql
 20240416145256_update_safety_privilege_label.up.sql
+20240502225801_add_hq_role.up.sql
 20240502175909_add_lookup_valid_columns_to_tac_and_loa_tables.up.sql
 20240502183613_add_support_for_standalone_payment_cap.up.sql
 20240503123556_add_diversion_reason_to_mto_shipments.up.sql
+20240502225801_add_hq_role.up.sql
```
Comment (ajlusk) [inline migrations/app/migrations_manifest.txt:931]: Looks like there is a duplicate here.

Comment (r-mettler) [reply migrations/app/migrations_manifest.txt]: wah wah, thanks git merge...it was just in the INT branch and not in the main one, fixed

**Comment by r-mettler** [comment]
> > Either way I create an HQ User something seems to be missing.
> > When I create an HQ user as the testing instructions suggest, via the admin app, I am unable to log in as the new user. I get a Forbidden error. The error message says the account is not active, however `users.active` is `true`. The user is missing a value in the `users.okta_id` column. When I manually added a uuid there the user worked as expected. This appears to happen for all users created in the admin app so could be an issue on my end.
> > When I create and HQ user from `http://officelocal:3000/devlocal-auth/login`, the user is only assigned just the TOO role (verified in the DB) rather than the HQ role or both. This method of user creation is not listed in the testing instructions but seemed relevant to test.
> 
> Its a known issue that the users are missing okta_id when created in the admin app. I resolved the Forbidden error though so you should be able to login.

We are generating fake okta ids now for accounts when created using admin app or requested using office app and on devlocal or officelocal.

**Review by ajlusk — APPROVED** [review]
Functioning as desired. I'm able to create a functioning HQ user via all three user creation methods (admin create, office create, office request -> admin approve). Switching roles works correctly.

This is a nitpick but I might move pkg/handlers/authentication/devlocal.go:135 up with the rest of the user types above the comment.

It's probably best to wait to merge until the `pkg/migrate/ListFiles.go`, `go.mod`, and `go.sum` are no longer listed as files changed/

**Comment by r-mettler** [comment]
> Functioning as desired. I'm able to create a functioning HQ user via all three user creation methods (admin create, office create, office request -> admin approve). Switching roles works correctly.
> 
> This is a nitpick but I might move pkg/handlers/authentication/devlocal.go:135 up with the rest of the user types above the comment.
> 
> ~It's probably best to wait to merge until the `pkg/migrate/ListFiles.go`, `go.mod`, and `go.sum` are no longer listed as files changed.~

Moved devlocal line 135 with the rest of the user types.

**Review by MInthavongsay — APPROVED** [review]
Works! There seems to be a styling issue with the layout width of the Request User view.
Screenshot 2024-05-22 at 2 28 44 PM

**Comment by r-mettler** [comment]
> Works! There seems to be a styling issue with the layout width of the Request User view. Screenshot 2024-05-22 at 2 28 44 PM

Yeah Alex and I noticed that as well, but it doesn't always happen. Something funky with how that page displays idk.

**Comment by ajlusk** [comment]
> Works! There seems to be a styling issue with the layout width of the Request User view. Screenshot 2024-05-22 at 2 28 44 PM

This happens on a lot of forms. It's been [discussed in slack](https://caci-bits.slack.com/archives/C05DX3AG6EL/p1711464129589109). Might be worth creating an issue for if there isn't already one. I haven't found one.

**Pull request merged**